velox/functions/prestosql/aggregates/sketch/tests/SfmSketchTest.cpp

```diff
@@ -0,0 +1,567 @@
+ public:
+  static void SetUpTestCase() {
+    memory::MemoryManager::testingSetInstance(memory::MemoryManager::Options{});
+  }
+
+  double mergeRandomizedResponseProbabilities(double p1, double p2) {
+    return (p1 + p2 - 3 * p1 * p2) / (1 - 2 * p1 * p2);
+  }
+
+  double calculateRandomizedResponseProbability(double epsilon) {
+    if (epsilon == std::numeric_limits<double>::infinity()) {
+      return 0.0;
+    }
+    return 1.0 / (1.0 + exp(epsilon));
+  }
+
+ protected:
+  int32_t numberOfBuckets_ = 4096;
+  int32_t precision_ = 24;
+  std::shared_ptr<memory::MemoryPool> pool_{
+      memory::memoryManager()->addLeafPool()};
+  HashStringAllocator allocator_{pool_.get()};
+};
+
+TEST_F(SfmSketchTest, computeIndex) {
+  std::vector<int32_t> indexBitLength = {6, 8, 10, 12};
+  for (auto& length : indexBitLength) {
+    const int32_t index = 5;
+    const auto hash = static_cast<uint64_t>(index) << (64 - length);
+    ASSERT_EQ(SfmSketch::computeIndex(hash, length), index);
+  }
+}
+
+TEST_F(SfmSketchTest, numberOfTrailingZeros) {
+  std::vector<int32_t> indexBitLength = {6, 8, 10, 12};
+  for (auto& length : indexBitLength) {
+    for (int32_t zeros = 0; zeros < 63; zeros++) {
+      const uint64_t hash = 1ULL << zeros;
+      const int32_t kBitWidth = sizeof(uint64_t) * 8;
+      const uint64_t value = hash | (1ULL << (kBitWidth - length));
+
+      const auto trailingZeros = static_cast<int32_t>(__builtin_ctzll(value));
+      ASSERT_EQ(trailingZeros, std::min(zeros, 64 - length));
+    }
+  }
+}
+
+TEST_F(SfmSketchTest, numberOfBuckets) {
+  std::vector<int32_t> indexBitLength = {6, 8, 10, 12};
+
+  for (auto& length : indexBitLength) {
+    ASSERT_EQ(SfmSketch::numBuckets(length), 1U << length);
+  }
+}
+
+TEST_F(SfmSketchTest, privacyEnabled) {
+  SfmSketch sketch(&allocator_);
+  sketch.initialize(numberOfBuckets_, precision_);
+  ASSERT_FALSE(sketch.privacyEnabled());
+  sketch.enablePrivacy(std::numeric_limits<double>::infinity());
+  ASSERT_FALSE(sketch.privacyEnabled());
+  sketch.enablePrivacy(1.0);
+  ASSERT_TRUE(sketch.privacyEnabled());
+  // Enable privacy twice will throw exception.
+  VELOX_ASSERT_THROW(sketch.enablePrivacy(1.0), "privacy is already enabled.");
+}
+
+TEST_F(SfmSketchTest, mergeNonPrivacy) {
+  auto sketchA = SfmSketch(1, &allocator_);
+  sketchA.initialize(numberOfBuckets_, precision_);
+  auto sketchB = SfmSketch(2, &allocator_);
+  sketchB.initialize(numberOfBuckets_, precision_);
+
+  // Add distinct values to the sketches
+  for (int i = 0; i < 1000; i++) {
+    sketchA.add(i);
+    sketchB.add(i + 1000);
+  }
+
+  auto refBitMapRange = sketchA.bits();
+  // Create a mutable copy for the test
+  std::vector<int8_t> refBitMap(refBitMapRange.begin(), refBitMapRange.end());
+  // Merge the sketches
+  sketchA.mergeWith(sketchB);
+  // Size of bitmap after merge should be the same
+  ASSERT_EQ(sketchA.numBits(), numberOfBuckets_ * precision_);
+
+  velox::bits::orBits(
```

Comment:
I feel this test is just repeats the code being tested. It makes it harder to maintain tests, but it doesn't really verify much. Consider removing.